chrome/content_script.js

```diff
@@ -891,14 +901,19 @@ chrome.runtime.onMessage.addListener(
       passwordalert.start_(msg);
     }
 );
-document.addEventListener('DOMContentLoaded', function() {
-  passwordalert.domContentLoaded_ = true;
-  passwordalert.completePageInitializationIfReady_();
-});
-// Check to see if we already missed DOMContentLoaded:
-if (document.readyState == 'interactive' ||
-    document.readyState == 'complete' ||
-    document.readyState == 'loaded') {
-  passwordalert.domContentLoaded_ = true;
-  passwordalert.completePageInitializationIfReady_();
+
+domReadyCheck = function() {
+  var loginForm = document.querySelector('#view_container > form');
+  if (loginForm && document.getElementById('Email')) {
+    passwordalert.domContentLoaded_ = true;
+  } else if (document.getElementById('hiddenEmail') &&
+             document.getElementsByName('password')){
+    passwordalert.domContentLoaded_ = true;
+  }
+  if (passwordalert.domContentLoaded_) {
+    window.removeEventListener('DOMNodeInserted', domReadyCheck);
+    passwordalert.completePageInitializationIfReady_();
+  }
 }
+
+window.addEventListener('DOMNodeInserted', domReadyCheck);
```

Comment:
Why listen for DOMNodeInserted instead of DOMContentLoaded?

Reply:
One the version of Google's login page where the user gets to select an account and is then prompted to enter their password, DOMContentLoaded triggers on the page where the user gets to select an account, but not again when they enter their password. In our testing, in order for completePageInitializationIfReady_() to be called when a password form exists on the page, we had to listen for DOM change events on check if the login form exists. I'm sure you know how Google's login page operates better than we do but our hypothesis is that the page only loads once, and is rewritten along with the url as the user steps though the login process. We more than welcome a better way to trigger an action when a user lands on the password prompt but we could not find one in our limited testing.

Reply:
That makes sense and I see the same behavior on the account selection page. Thanks for noticing this and the explanation!

I'll put this behind a check for accounts.google.com URLs so that we do not cause performance issues by doing DOM event listening on every webpage. It also appears that DOMNodeInserted is deprecated, so I'll use a MutationObserver: https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Mutation_events